src/components/Checkbox/Checkbox.st.css

```diff
@@ -95,7 +109,44 @@
     color: value(CurrentTextColor);
 }
 
+.suffix {
+    margin-left: 5px;
+}
+
+.box .suffix {
+    margin-left: auto;
+}
+
 /* STATES */
+.box.disabled {
+    cursor: default;
+}
+
+.box:hover, .box:focus-within {
+    border-color: applyOpacity(value(CurrentBorderColor), 1);
+}
+
+.box.disabled:hover:not(.checked), .box.disabled:focus-within:not(.checked) {
+    border-color: applyOpacity(value(CurrentBorderColor), 0.6);
+}
+
+.box.checked {
+    background-color: applyOpacity(value(CurrentIconColor), 0.1);
+    border-color: value(CurrentIconColor);
+}
+
+.box.checked:not(.disabled):hover, .box.checked:not(.disabled):focus-within {
+    border-color: applyOpacity(value(CurrentIconColor), 0.6);
+}
+
+.box:not(.disabled):hover .icon {
+    border-color: applyOpacity(value(CurrentBorderColor), 1);
+}
```

Comment:
The correct syntax for using Stylable states is like this:
```css
.root:disabled {
  ...
}
```

So you'll need to adjust this code accordingly.
Not that the state is configured on the `.root` class, and I think it won't work on other classes.

So for instance I think it should be written like this:
```css
.root:disabled.box {
  ...
}
```